python/pyspark/rdd.py

```diff
@@ -84,7 +84,7 @@ def portable_hash(x):
         h ^= len(x)
         if h == -1:
             h = -2
-        return h
+        return int(h)
```

Comment:
Just asking the dumb question here, but is this intended to return an int? `sys.maxsize` does not appear to be the max positive 32-bit int.